packages/cli/src/openapi/util.js

```diff
@@ -1,24 +1,60 @@
 const { buildAppMap } = require('@appland/models');
 
 function classNameToOpenAPIType(className) {
-  let typeName = className;
-  if (!typeName || typeName === '') {
+  if (!className || className === '') {
     return 'unknown';
   }
-  typeName = typeName.toLowerCase();
 
-  switch (typeName) {
-    case 'hash':
-    case 'activesupport::hashwithindifferentaccess':
-      return 'object';
-    case 'nilclass':
-      return 'string';
-    case 'trueclass':
-    case 'falseclass':
-      return 'boolean';
-    default:
-      return typeName;
-  }
+  const mapRubyType = (t) => {
+    switch (t) {
+      case 'array':
+        return 'array';
+      case 'hash':
+      case 'activesupport::hashwithindifferentaccess':
+        return 'object';
+      case 'nilclass':
+        return 'string';
+      case 'trueclass':
+      case 'falseclass':
+        return 'boolean';
+      default:
+        return undefined;
+    }
+  };
+
+  const mapPythonType = (t) => {
+    if (!t.startsWith('builtins.')) {
+      return undefined;
+    }
+
+    switch (t.substr(9)) {
+      case 'bool':
+        return 'boolean';
+      case 'dict':
+        return 'object';
+      case 'int':
+        return 'integer';
+      case 'list':
+        return 'array';
+      case 'str':
+        return 'string';
+      default:
+        return undefined;
+    }
+  };
+
+  const mapJavaType = (t) => {
+    switch (t) {
+      case 'java.lang.string':
+        return 'string';
+      default:
+        return undefined;
+    }
+  };
+
+  const mapper = (t) => mapRubyType(t) || mapPythonType(t) || mapJavaType(t);
+
+  return mapper(className.toLowerCase()) || className;
```

Comment:
@kgilpin @ptrdvrk The previous version of this code also just returns the class name if no other mapping is found. But, is this really the behavior we want? At the very least, it seems like we should also log a warning that we're generating an invalid spec.